tools/src/main/java/it/eng/tools/model/Serializer.java

```diff
@@ -0,0 +1,178 @@
+		try {
+			return jsonMapperPlain.writeValueAsString(toSerialize);
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+
+	/**
+	 * Convert object to jsonNode, without annotations. Used in tests
+	 * @param toSerialize
+	 * @return JsonNode
+	 */
+	public static JsonNode serializePlainJsonNode(Object toSerialize) {
+		return jsonMapperPlain.convertValue(toSerialize, JsonNode.class);
+	}
+
+	/**
+	 * Converts json string (plain) to java object
+	 * @param <T> Type of class
+	 * @param jsonStringPlain json string
+	 * @param clazz
+	 * @return Java object converted from json
+	 */
+	public static <T> T deserializePlain(String jsonStringPlain, Class<T> clazz) {
+		try {
+			T obj = jsonMapperPlain.readValue(jsonStringPlain, clazz);
+			Set<ConstraintViolation<T>> violations = validator.validate(obj);
+			if(violations.isEmpty()) {
+				return obj;
+			}
+			throw new ValidationException(
+					violations
+					.stream()
+					.map(v -> v.getPropertyPath() + " " + v.getMessage())
+					.collect(Collectors.joining(",")));
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+
+	/**
+	 * Serialize java object to json compliant with Dataspace protocol (contains prefixes for json fields)
+	 * @param toSerialize java object to serialize
+	 * @return Json string - with Dataspace prefixes
+	 */
+	public static String serializeProtocol(Object toSerialize) {
+		try {
+			return jsonMapper.writeValueAsString(toSerialize);
+		} catch (JsonProcessingException e) {
+			e.printStackTrace();
+		}
+		return null;
+	}
+
+	/**
+	 * Convert object to JsonNode with prefixes. Used in tests
+	 * @param toSerialize
+	 * @return
+	 */
+	public static JsonNode serializeProtocolJsonNode(Object toSerialize) {
+		return jsonMapper.convertValue(toSerialize, JsonNode.class);
+	}
+
+	public static List<JsonNode> serializeProtocolListOfJsonNode(Object toSerialize) {
+		return jsonMapper.convertValue(toSerialize, List.class);
+	}
+	
+	/**
+	 * Convert Dataspace json (with prefixes) to java object, performs validation for @context and @type before converting to java
+	 * Enforce validation for mandatory fields
+	 * @param <T>
+	 * @param jsonNode
+	 * @param clazz
+	 * @return
+	 */
+	public static <T> T deserializeProtocol(JsonNode jsonNode, Class<T> clazz) {
+		validateProtocol(jsonNode, clazz);
+		T obj = jsonMapper.convertValue(jsonNode, clazz);
+		Set<ConstraintViolation<T>> violations = validator.validate(obj);
+		if(violations.isEmpty()) {
+			return obj;
+		}
+		throw new ValidationException(
+				violations
+				.stream()
+				.map(v -> v.getPropertyPath() + " " + v.getMessage())
+				.collect(Collectors.joining(",")));
+	}
+
+	/**
+	 * Checks for @context and @type if present and if values are correct
+	 * @param <T>
+	 * @param jsonNode
+	 * @param clazz
+	 * @throws jakarta.validationException 
+	 */
+	private static <T> void validateProtocol(JsonNode jsonNode, Class<T> clazz) {
+		/*
```

Comment:
If this commented code is not needed, please delete it (and method in same case)